### PR TITLE
[form] Add <SelectRow>

### PR DESCRIPTION
**Pull request by zhusee2**
### Purpose
Add a row-based component which lets you to pick options from a `<SelectList>` rendered inside a `<Popover>`

### Implement
- Add new `<SelectRow>`, which takes `<SelectOption>` as children and renders `<SelectList>` inside a popover when user clicks on the row.

### CODE REVIEW

**Comment by zhusee2** [comment]
Coverall 好嚴格...!!!

**Comment by cjies** [comment]
Or we set a acceptable decrease threshold for failure?
screen shot 2017-10-26 at 3 32 06 pm

Like if coverage drop to 70% will cause failure or drop 1% in each PR will cause failure too XD

**Comment by zhusee2** [comment]
Sounds good. Let's set threshold to 85% and decrease threshold to 1%?

File: packages/form/src/SelectRow.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+        };
+
+        return (
+            <ListRow className={wrapperClassName} {...rowProps}>
+                <Content minified={false} disabled={disabled} {...contentProps}>
+                    <Text
+                        ref={(ref) => { this.anchorNode = ref; }}
+                        bold={!ineditable}
+                        basic={label} />
+                    <Icon type="unfold" />
```
Comment (cjies) [inline packages/form/src/SelectRow.js]: The unfold icon looks more suitable for the anchor point? The popover placed at the bottom of text is a little bit weird, or is it a design spec?

File: packages/core/src/mixins/anchored.js [hunk 2]
```diff
@@ -207,26 +207,23 @@ const anchored = ({
...
-                        arrowSafeAreaWidth
-                    );
+                    nextState.arrowPosition.left = anchorHalfWidth;
                     break;
             }
 
+            // Calibrate arrow position so it stays in safe area
+            if (nextState.arrowPosition.left < edgePadding
+                || nextState.arrowPosition.left > arrowSafeAreaWidth) {
+                delete nextState.arrowPosition;
```
Comment (cjies) [inline packages/core/src/mixins/anchored.js]: If the anchored component place in the right edge, the `arrowPosition` will be removed.

If remove calculated arrow position: 
screen shot 2017-10-26 at 5 30 45 pm

Remains arrow position:
screen shot 2017-10-26 at 5 30 30 pm

Comment (cjies) [reply packages/core/src/mixins/anchored.js]: If revert the `arrowPosition`, the arrow float to right when the popover append on <Text> component XDD

screen shot 2017-10-26 at 5 36 02 pm

**Comment by zhusee2** [comment]
Let me adjust them for that

**Comment by zhusee2** [comment]
After a quick discussion with kouko, I'm going to use the `<Icon>` as anchor so the `<Popover>` is placed on the right side. I'll also fix arrow position for `anchored()` mixin.

But in this case I'll keep the arrow position as-is, because the whole row is the triggering area and it makes less sense to move the arrow to the `<Icon>`.

**Comment by zhusee2** [comment]
### Updated screenshot

**Comment by zhusee2** [comment]
Rebased with latest `develop`

**Pull request merged**